### PR TITLE
Add new test for linear id

### PR DESCRIPTION
**Pull request by ole-alb**
**_Describe your changes here:_**

Test if the element created by the set_linear_id function is correct and rename the linear id tests to `t8_gtest_set_linear_id` and `t8_gtest_get_linear_id`. 

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/DLR-AMR/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### License

- [ ] The author added a BSD statement to `doc/` (or already has one)

#### Tag Label

- [ ] The author added the patch/minor/major label in accordance to semantic versioning.

### CODE REVIEW

**Comment by Davknapp** [comment]
Just had a quick look at it. Maybe it is a good idea to name the tests after the function that they are testing. So could you rename your file to t8_element_set_linear_id? And the init liniear id test to t8_element_get_linear_id? Thank you!

File: test/t8_schemes/t8_gtest_set_linear_id.cxx [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_set_linear_id.cxx:6]:
```suggestion
  Copyright (C) 2025 the developers
```

File: test/t8_schemes/t8_gtest_set_linear_id.cxx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+/** In this test we iterate through all elements. 
+ * For every leaf we check if the element is equal to the element we get when setting it from the linear id.
+ * The id_counter is then increased to match the id of the next leaf.
+ */
+class class_test_set_linear_id: public TestDFS {
+  void
+  check_element () override
+  {
+    t8_debugf ("maxlvl: %i\n", maxlvl);
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_set_linear_id.cxx:38]:
```suggestion
```

File: test/t8_schemes/t8_gtest_set_linear_id.cxx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    t8_debugf ("level: %d\n", scheme->element_get_level (eclass, element));
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_set_linear_id.cxx:39]:
```suggestion
```

File: test/t8_schemes/t8_gtest_set_linear_id.cxx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    if (scheme->element_get_level (eclass, element) == maxlvl) {
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_set_linear_id.cxx:40]: Why do we only test on maxlvl? We could also test on every level until we hit the maximum level.

File: test/t8_schemes/t8_gtest_set_linear_id.cxx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  {
+    /* Destroy element */
+    scheme->element_destroy (eclass, 1, &test_element);
+
+    /* Destroy DFS test */
+    dfs_test_teardown ();
+  }
+  t8_element_t *test_element;
+  int id_counter = 0;
+  int maxlvl;
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_set_linear_id.cxx:84]:
```suggestion
#if T8_ENABLE_LESS_TESTS
  const int maxlvl = 3;
#else 
  const int maxlvl = 5;
#endif
```

File: test/t8_schemes/t8_gtest_set_linear_id.cxx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  void
+  SetUp () override
+  {
+    dfs_test_setup ();
+    /* Get element and initialize it */
+    scheme->element_new (eclass, 1, &test_element);
+    scheme->element_copy (eclass, element, test_element);
+    maxlvl = set_maxlvl ();
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_set_linear_id.cxx:71]:
```suggestion
```

File: test/t8_schemes/t8_gtest_set_linear_id.cxx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+      id_counter++;
+      EXPECT_ELEM_EQ (scheme, eclass, element, test_element);
+    }
+    else {
+      return;
+    }
+  }
+
+ protected:
+  int
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_set_linear_id.cxx:52]: We don't need this function, see my comment down below.

**Review by Davknapp — CHANGES_REQUESTED** [review]

**Review by Davknapp — APPROVED** [review]
